packages/guides/src/Nodes/AbstractNode.php

```diff
@@ -89,6 +89,11 @@ public function withOptions(array $options): Node
         return $result;
     }
 
+    public function addOption(string $identifier, int|bool|string|null $option): void
```

Comment:
I think we should keep the `withOptions` method and not introduce a new method to set options. The immutable behavior of this class was intended to prevent users from changing the nodes by accident. And make it easier to see when the cache should be invalidated.

Reply:
So, we can add and remove nodes in the nodestransformers, but we cannot change them?

Reply:
Add, replace and remove, if you return a new node the old one will be replaced.